contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -0,0 +1,959 @@
+            _wethSharePct <= allowedWethShareEnd
+        ) {
+            emit PoolRebalanced(_wethSharePct);
+        } else {
+            revert PoolRebalanceOutOfBounds(
+                _wethSharePct,
+                allowedWethShareStart,
+                allowedWethShareEnd
+            );
+        }
+    }
+
+    /**
+     * Burns any OETHb tokens remaining on the strategy contract
+     */
+    function _burnOethbOnTheContract() internal {
+        uint256 _oethbBalance = IERC20(OETHb).balanceOf(address(this));
+        if (_oethbBalance > 0) {
+            IVault(vaultAddress).burnForStrategy(_oethbBalance);
+        }
+    }
+
+    /// @dev this function assumes there are no uncollected tokens in the clPool owned by the.
+    ///      strategy contract. For that reason any liquidity withdrawals also collect the tokens.
+    function _updateUnderlyingAssets() internal {
+        if (tokenId == 0) {
+            underlyingAssets = 0;
+            emit UnderlyingAssetsUpdated(underlyingAssets);
+            return;
+        }
+
+        uint128 _liquidity = _getLiquidity();
+
+        /**
+         * Our net value represent the smallest amount of tokens we are able to extract from the position
+         * given our liquidity.
+         *
+         * The least amount of tokens extraditable from the position is where the active trading price is
+         * at the ticker 0 meaning the pool is offering 1:1 trades between WETH & OETHb. At that moment the pool
+         * consists completely of OETHb and no WETH.
+         *
+         * The more swaps from WETH -> OETHb happen on the pool the more the price starts to move towards the -1
+         * ticker making OETHb (priced in WETH) more expensive.
+         */
+        (uint256 _wethAmount, uint256 _oethbAmount) = helper
+            .getAmountsForLiquidity(
+                sqrtRatioX96TickClosestToParity, // sqrtRatioX96
+                sqrtRatioX96TickLower, // sqrtRatioAX96
+                sqrtRatioX96TickHigher, // sqrtRatioBX96
+                _liquidity
+            );
+
+        require(_wethAmount == 0, "Non zero wethAmount");
+        underlyingAssets = _oethbAmount;
+        emit UnderlyingAssetsUpdated(underlyingAssets);
+    }
+
+    /**
+     * @notice Deposit WETH to the strategy contract. This function does not add liquidity to the
+     *         underlying Aerodrome pool.
+     */
+    function depositAll() external override onlyVault nonReentrant {
+        uint256 _wethBalance = IERC20(WETH).balanceOf(address(this));
+        if (_wethBalance > 0) {
+            _deposit(WETH, _wethBalance);
+        }
+    }
+
+    /**
+     * @notice Withdraw an `amount` of assets from the platform and
+     *         send to the `_recipient`.
+     * @param _recipient  Address to which the asset should be sent
+     * @param _asset      WETH address
+     * @param _amount     Amount of WETH to withdraw
+     */
+    function withdraw(
```

Comment:
⚪️ It's kind of weird that we usually put our public methods first, then the internal ones, but this contract does rebalance, and then the rest of the external methods starting at line 747. Might be nice to move these (and their tiny `_deposit` and `_withdraw` methods up.

Reply:
Checking our codebase there are multiple instances where public and their corresponding internal functions are placed closely together rather than ordering functions by visiblity. Some examples:
- [AaveStrategy](https://github.com/OriginProtocol/origin-dollar/blob/master/contracts/contracts/strategies/AaveStrategy.sol#L63-L83)
- [CompoundStrategy](https://github.com/OriginProtocol/origin-dollar/blob/master/contracts/contracts/strategies/CompoundStrategy.sol#L82-L101)
- [ConvexEthMetaStrategy](https://github.com/OriginProtocol/origin-dollar/blob/master/contracts/contracts/strategies/ConvexEthMetaStrategy.sol#L168-L178)
- [FraxEthStrategy](https://github.com/OriginProtocol/origin-dollar/blob/master/contracts/contracts/strategies/FraxETHStrategy.sol#L48-L108)
- [Generalized4262Strategy](https://github.com/OriginProtocol/origin-dollar/blob/master/contracts/contracts/strategies/Generalized4626Strategy.sol#L61-L82)

IMO, it is easier to have public functions and the internal functions they call more closely together for convenience and readability. Though `depositAll` was out of place in the contract and moved: https://github.com/OriginProtocol/origin-dollar/pull/2184/commits/abb1300bd2e36e05f8e99aea5047483bf7b6741f

Reply:
Ok